From c1a904dedbf2f1333f139da71fdaa1ced4de80be Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Thu, 29 Mar 2018 14:01:43 +0800
Subject: [PATCH 10/11] build shared libs

Upstream-Status: Pending

Port patches from Fedora to build shared libs rather than static libs.

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase to 4.4.1

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

squash! build shared libs
---
 client/Makefile.am       |  7 ++-----
 common/tests/Makefile.am | 35 ++++++++++-------------------------
 configure.ac             | 12 ++----------
 dhcpctl/Makefile.am      | 18 ++++++------------
 omapip/Makefile.am       | 11 ++++-------
 relay/Makefile.am        |  7 ++-----
 server/Makefile.am       |  9 +++------
 server/tests/Makefile.am |  9 +++------
 8 files changed, 32 insertions(+), 76 deletions(-)

diff --git a/client/Makefile.am b/client/Makefile.am
index d1f9c15..10844f3 100644
--- a/client/Makefile.am
+++ b/client/Makefile.am
@@ -13,10 +13,7 @@ dhclient_SOURCES = $(srcdir)/client_tables.c $(srcdir)/clparse.c $(srcdir)/dhcli
 		   scripts/bsdos scripts/freebsd scripts/linux scripts/macos \
 		   scripts/netbsd scripts/nextstep scripts/openbsd \
 		   scripts/solaris scripts/openwrt
-dhclient_LDADD = ../common/libdhcp.@A@ ../omapip/libomapi.@A@ \
-		 @BINDLIBIRSDIR@/libirs.@A@ \
-		 @BINDLIBDNSDIR@/libdns.@A@ \
-		 @BINDLIBISCCFGDIR@/libisccfg.@A@ \
-		 @BINDLIBISCDIR@/libisc.@A@
+dhclient_LDADD = ../common/libdhcp.@A@ ../omapip/libomapi.la \
+		 -L$(BINDDIR) -lirs -ldns -lisccfg -lisc
 man_MANS = dhclient.8 dhclient-script.8 dhclient.conf.5 dhclient.leases.5
 EXTRA_DIST = $(man_MANS)
diff --git a/common/tests/Makefile.am b/common/tests/Makefile.am
index 322f77e..917120b 100644
--- a/common/tests/Makefile.am
+++ b/common/tests/Makefile.am
@@ -13,43 +13,28 @@ ATF_TESTS += alloc_unittest dns_unittest misc_unittest ns_name_unittest \
 
 alloc_unittest_SOURCES = test_alloc.c $(top_srcdir)/tests/t_api_dhcp.c
 alloc_unittest_LDADD = $(ATF_LDFLAGS)
-alloc_unittest_LDADD += ../libdhcp.@A@ ../../omapip/libomapi.@A@ \
-	@BINDLIBIRSDIR@/libirs.@A@ \
-	@BINDLIBDNSDIR@/libdns.@A@ \
-	@BINDLIBISCCFGDIR@/libisccfg.@A@  \
-	@BINDLIBISCDIR@/libisc.@A@
+alloc_unittest_LDADD += ../libdhcp.@A@ ../../omapip/libomapi.la \
+	-L$(BINDDIR) -lirs -ldns -lisccfg -lisc
 
 dns_unittest_SOURCES = dns_unittest.c $(top_srcdir)/tests/t_api_dhcp.c
 dns_unittest_LDADD = $(ATF_LDFLAGS)
-dns_unittest_LDADD += ../libdhcp.@A@ ../../omapip/libomapi.@A@ \
-	@BINDLIBIRSDIR@/libirs.@A@ \
-	@BINDLIBDNSDIR@/libdns.@A@ \
-	@BINDLIBISCCFGDIR@/libisccfg.@A@  \
-	@BINDLIBISCDIR@/libisc.@A@
+dns_unittest_LDADD += ../libdhcp.@A@  ../../omapip/libomapi.la \
+	-L$(BINDDIR) -lirs -ldns -lisccfg -lisc
 
 misc_unittest_SOURCES = misc_unittest.c $(top_srcdir)/tests/t_api_dhcp.c
 misc_unittest_LDADD = $(ATF_LDFLAGS)
-misc_unittest_LDADD += ../libdhcp.@A@ ../../omapip/libomapi.@A@ \
-	@BINDLIBIRSDIR@/libirs.@A@ \
-	@BINDLIBDNSDIR@/libdns.@A@ \
-	@BINDLIBISCCFGDIR@/libisccfg.@A@  \
-	@BINDLIBISCDIR@/libisc.@A@
+misc_unittest_LDADD += ../libdhcp.@A@  ../../omapip/libomapi.la \
+	-L$(BINDDIR) -lirs -ldns -lisccfg -lisc
 
 ns_name_unittest_SOURCES = ns_name_test.c $(top_srcdir)/tests/t_api_dhcp.c
 ns_name_unittest_LDADD = $(ATF_LDFLAGS)
-ns_name_unittest_LDADD += ../libdhcp.@A@ ../../omapip/libomapi.@A@ \
-	@BINDLIBIRSDIR@/libirs.@A@ \
-	@BINDLIBDNSDIR@/libdns.@A@ \
-	@BINDLIBISCCFGDIR@/libisccfg.@A@  \
-	@BINDLIBISCDIR@/libisc.@A@
+ns_name_unittest_LDADD += ../libdhcp.@A@  ../../omapip/libomapi.la \
+	-L$(BINDDIR) -lirs -ldns -lisccfg -lisc
 
 option_unittest_SOURCES = option_unittest.c $(top_srcdir)/tests/t_api_dhcp.c
 option_unittest_LDADD = $(ATF_LDFLAGS)
-option_unittest_LDADD += ../libdhcp.@A@ ../../omapip/libomapi.@A@ \
-	@BINDLIBIRSDIR@/libirs.@A@ \
-	@BINDLIBDNSDIR@/libdns.@A@ \
-	@BINDLIBISCCFGDIR@/libisccfg.@A@  \
-	@BINDLIBISCDIR@/libisc.@A@
+option_unittest_LDADD += ../libdhcp.@A@  ../../omapip/libomapi.la \
+	-L$(BINDDIR) -lirs -ldns -lisccfg -lisc
 
 check: $(ATF_TESTS)
 	@if test $(top_srcdir) != ${top_builddir}; then \
diff --git a/configure.ac b/configure.ac
index 2bf434d..97fc39c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -60,16 +60,8 @@ AC_SUBST(BINDCONFIG)
 # Use this to define _GNU_SOURCE to pull in the IPv6 Advanced Socket API.
 AC_USE_SYSTEM_EXTENSIONS
 
-AC_PROG_RANLIB
-
-AC_PATH_PROG(AR, ar)
-AC_SUBST(AR)
-
-if test "X$AR" = "X"; then
-	AC_MSG_ERROR([
-ar program not found.  Please fix your PATH to include the directory in
-which ar resides, or set AR in the environment with the full path to ar.])
-fi
+# Use libtool to simplify building of shared libraries
+AC_PROG_LIBTOOL
 
 AC_CONFIG_HEADERS([includes/config.h])
 
diff --git a/dhcpctl/Makefile.am b/dhcpctl/Makefile.am
index 0d66971..5b77e90 100644
--- a/dhcpctl/Makefile.am
+++ b/dhcpctl/Makefile.am
@@ -4,23 +4,17 @@ BINDLIBISCCFGDIR=@BINDLIBISCCFGDIR@
 BINDLIBISCDIR=@BINDLIBISCDIR@
 
 bin_PROGRAMS = omshell
-lib_LIBRARIES = libdhcpctl.a
+lib_LTLIBRARIES = libdhcpctl.la
 noinst_PROGRAMS = cltest
 man_MANS = omshell.1 dhcpctl.3
 EXTRA_DIST = $(man_MANS)
 
 omshell_SOURCES = omshell.c
-omshell_LDADD = libdhcpctl.a ../common/libdhcp.a ../omapip/libomapi.a \
-		$(BINDLIBIRSDIR)/libirs.a \
-		$(BINDLIBDNSDIR)/libdns.a \
-	        $(BINDLIBISCCFGDIR)/libisccfg.a \
-		$(BINDLIBISCDIR)/libisc.a
+omshell_LDADD = libdhcpctl.la ../common/libdhcp.a ../omapip/libomapi.la \
+               -L$(BINDDIR) -lirs -ldns -lisccfg -lisc
 
-libdhcpctl_a_SOURCES = dhcpctl.c callback.c remote.c
+libdhcpctl_la_SOURCES = dhcpctl.c callback.c remote.c
 
 cltest_SOURCES = cltest.c
-cltest_LDADD = libdhcpctl.a ../common/libdhcp.a ../omapip/libomapi.a \
-	       $(BINDLIBIRSDIR)/libirs.a \
-	       $(BINDLIBDNSDIR)/libdns.a \
-	       $(BINDLIBISCCFGDIR)/libisccfg.a \
-	       $(BINDLIBISCDIR)/libisc.a
+cltest_LDADD = libdhcpctl.la ../common/libdhcp.a ../omapip/libomapi.la \
+              -L$(BINDDIR) -lirs -ldns -lisccfg -lisc
diff --git a/omapip/Makefile.am b/omapip/Makefile.am
index 5b61581..9a40f1e 100644
--- a/omapip/Makefile.am
+++ b/omapip/Makefile.am
@@ -3,10 +3,10 @@ BINDLIBDNSDIR=@BINDLIBDNSDIR@
 BINDLIBISCCFGDIR=@BINDLIBISCCFGDIR@
 BINDLIBISCDIR=@BINDLIBISCDIR@
 
-lib_LIBRARIES = libomapi.a
+lib_LTLIBRARIES = libomapi.la
 noinst_PROGRAMS = svtest
 
-libomapi_a_SOURCES = protocol.c buffer.c alloc.c result.c connection.c \
+libomapi_la_SOURCES = protocol.c buffer.c alloc.c result.c connection.c \
 		       errwarn.c listener.c dispatch.c generic.c support.c \
 		       handle.c message.c convert.c hash.c auth.c inet_addr.c \
 		       array.c trace.c toisc.c iscprint.c isclib.c
@@ -15,8 +15,5 @@ man_MANS = omapi.3
 EXTRA_DIST = $(man_MANS)
 
 svtest_SOURCES = test.c
-svtest_LDADD = libomapi.a \
-	       $(BINDLIBIRSDIR)/libirs.a \
-	       $(BINDLIBDNSDIR)/libdns.a \
-	       $(BINDLIBISCCFGDIR)/libisccfg.a \
-	       $(BINDLIBISCDIR)/libisc.a
+svtest_LDADD = libomapi.la \
+		-L$(BINDDIR) -lirs -ldns -lisccfg -lisc
diff --git a/relay/Makefile.am b/relay/Makefile.am
index 2ba5979..2c1fb38 100644
--- a/relay/Makefile.am
+++ b/relay/Makefile.am
@@ -2,11 +2,8 @@ AM_CPPFLAGS = -DLOCALSTATEDIR='"@localstatedir@"'
 
 sbin_PROGRAMS = dhcrelay
 dhcrelay_SOURCES = dhcrelay.c
-dhcrelay_LDADD = ../common/libdhcp.@A@ ../omapip/libomapi.@A@ \
-		 @BINDLIBIRSDIR@/libirs.@A@ \
-		 @BINDLIBDNSDIR@/libdns.@A@ \
-		 @BINDLIBISCCFGDIR@/libisccfg.@A@ \
-		 @BINDLIBISCDIR@/libisc.@A@
+dhcrelay_LDADD = ../common/libdhcp.@A@ ../omapip/libomapi.la \
+		 -L$(BINDDIR) -lirs -ldns -lisccfg -lisc
 man_MANS = dhcrelay.8
 EXTRA_DIST = $(man_MANS)
 
diff --git a/server/Makefile.am b/server/Makefile.am
index 787efca..d533715 100644
--- a/server/Makefile.am
+++ b/server/Makefile.am
@@ -13,12 +13,9 @@ dhcpd_SOURCES = dhcpd.c dhcp.c bootp.c confpars.c db.c class.c failover.c \
 		dhcpv6.c mdb6.c ldap.c ldap_casa.c leasechain.c ldap_krb_helper.c
 
 dhcpd_CFLAGS = $(LDAP_CFLAGS)
-dhcpd_LDADD = ../common/libdhcp.@A@ ../omapip/libomapi.@A@ \
-	      ../dhcpctl/libdhcpctl.@A@ \
-	      $(BINDLIBIRSDIR)/libirs.@A@ \
-	      $(BINDLIBDNSDIR)/libdns.@A@ \
-	      $(BINDLIBISCCFGDIR)/libisccfg.@A@ \
-	      $(BINDLIBISCDIR)/libisc.@A@ $(LDAP_LIBS)
+dhcpd_LDADD = ../common/libdhcp.@A@ ../omapip/libomapi.la \
+	      ../dhcpctl/libdhcpctl.la -L$(BINDDIR) \
+	      -lirs -ldns -lisccfg -lisc $(LDAP_LIBS)
 
 man_MANS = dhcpd.8 dhcpd.conf.5 dhcpd.leases.5
 EXTRA_DIST = $(man_MANS)
diff --git a/server/tests/Makefile.am b/server/tests/Makefile.am
index 699a0b6..78e81c6 100644
--- a/server/tests/Makefile.am
+++ b/server/tests/Makefile.am
@@ -18,12 +18,9 @@ DHCPSRC = ../dhcp.c ../bootp.c ../confpars.c ../db.c ../class.c      \
           ../ldap.c ../ldap_casa.c ../dhcpd.c ../leasechain.c
 
 DHCPLIBS = $(top_builddir)/common/libdhcp.@A@ \
-	  $(top_builddir)/omapip/libomapi.@A@ \
-          $(top_builddir)/dhcpctl/libdhcpctl.@A@ \
-	  $(BINDLIBIRSDIR)/libirs.@A@ \
-	  $(BINDLIBDNSDIR)/libdns.@A@ \
-	  $(BINDLIBISCCFGDIR)/libisccfg.@A@ \
-	  $(BINDLIBISCDIR)/libisc.@A@
+	  $(top_builddir)/omapip/libomapi.la \
+	  $(top_builddir)/dhcpctl/libdhcpctl.la \
+	  -L$(BINDDIR) -lirs -ldns -lisccfg -lisc
 
 ATF_TESTS =
 if HAVE_ATF
-- 
1.8.3.1

